transport/tchannel/handler.go

```diff
@@ -244,12 +254,17 @@ func (rw *responseWriter) Write(s []byte) (int, error) {
 }
 
 func (rw *responseWriter) Close() error {
-	var err error
+	err := rw.ensureWroteHeaders()
+
 	if rw.bodyWriter != nil {
-		err = rw.bodyWriter.Close()
+		if closeErr := rw.bodyWriter.Close(); err == nil {
```

Comment:
this caught me off guard since normally when we inline an error check it's the error we got from the function we ran.

Since getting the closeErr and checking if err==nil, can we make these two separate lines?
```
closeErr := rw.bodyWriter.Close()
if err == nil {
  err = closeErr
}
```